diff -NaurBwd a/src/GNUmakefile b/src/GNUmakefile
--- a/src/GNUmakefile	2017-05-10 15:08:48.672212014 +0700
+++ b/src/GNUmakefile	2017-05-10 15:10:23.462693220 +0700
@@ -150,7 +150,7 @@
 	$$(MKDIR) -p $$(@D)
 	$$(CP) -p $$< $$@
 
-$$(DESTDIR)$(3)/openssl/ssl/%.lua: $$(d)/openssl.ssl.%.lua
+$$(DESTDIR)$(3)/openssl/ocsp/%.lua: $$(d)/openssl.ocsp.%.lua
 	$$(LUAC$(1)_$(d)) -p $$<
 	$$(MKDIR) -p $$(@D)
 	$$(CP) -p $$< $$@

