diff -NaurBwd a/src/unix.c b/src/unix.c
--- a/src/unix.c	2017-05-10 18:12:19.819388720 +0700
+++ b/src/unix.c	2017-05-10 18:13:55.792661605 +0700
@@ -483,6 +483,7 @@
 } /* lua_absindex() */
 
 
+#ifndef LUA_OK
 static void *luaL_testudata(lua_State *L, int index, const char *tname) {
 	void *p = lua_touserdata(L, index);
 	int eq;
@@ -523,6 +524,7 @@
 
 #define luaL_newlib(L, l) \
 	(luaL_newlibtable((L), (l)), luaL_setfuncs((L), (l), 0))
+#endif
 
 
 #define lua_rawlen lua_objlen

